### PR TITLE
Handle Envoy 1.24 Deprecations

### LINKED ISSUE #4802: Handle Envoy 1.24.0 deprecations
See: https://www.envoyproxy.io/docs/envoy/latest/version_history/v1.24/v1.24.0#deprecated

Assuming 1.24.0 will be picked up in Contour 1.23.0, we should make these changes in Contour 1.24.0 (to ensure a smooth upgrade), since they seem to include deprecations paired with additions to the now-preferred fields

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Fixes #4802

See: https://www.envoyproxy.io/docs/envoy/latest/version_history/v1.24/v1.24.0#deprecated

- Bump go-control-plane to latest version for Envoy 1.24.0 (Commit: https://github.com/envoyproxy/go-control-plane/commit/799a7af9e5b9b3c492642319bf6a71cdfffc9cac)
- vhost.Cors -> typed filter config
- No longer need to provide total weight of a set of weighted clusters
- Change header append to append action
  - set header -> OVERWRITE_IF_EXISTS_OR_ADD (previous append false)
  - add header -> APPEND_IF_EXISTS_OR_ADD (previous append true)

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
upgrade tests failed w/ < 90% request success, re-running in CI and locally

File: internal/featuretests/v3/routeweight_test.go [hunk 1]
```diff
@@ -659,14 +659,11 @@ func routeweightedcluster(clusters ...weightedcluster) *envoy_route_v3.Route_Rou
 
 func weightedclusters(clusters []weightedcluster) *envoy_route_v3.WeightedCluster {
```
Comment (skriss) [inline internal/featuretests/v3/routeweight_test.go:660]: not directly related to your PR but this looks identical to `weightedClusters` in `internal/featuretests/v3/envoy.go`, would be nice to remove this one.

**Review by skriss — APPROVED** [review]
LGTM, one little cleanup item for here or a follow-up.

File: internal/featuretests/v3/routeweight_test.go (hunk 1 above)
Comment (sunjayBhatia) [reply internal/featuretests/v3/routeweight_test.go]: done!

**Pull request merged**